src/plugins/http/src/httpServer.c

```diff
@@ -400,7 +400,7 @@ static bool httpReadData(HttpContext *pContext) {
     } else if (nread < 0) {
       if (errno == EINTR || errno == EAGAIN || errno == EWOULDBLOCK) {
         httpDebug("context:%p, fd:%d, read from socket error:%d, wait another event", pContext, pContext->fd, errno);
-        continue;  // later again
+        return false;
```

Comment:
Excellent!

Reply:
@jianqingdu I have discussed with @dapan1121 and we drew a conclusion: if the errno is **EINTR**, the read should immediately continue, while if the errno is EAGAIN or EWOULDBLOCK, return false.

Reply:
@winshining I agree with you, and make the change,  please recheck the PR